utils/file-logging/src/lib.rs

```diff
@@ -61,14 +62,85 @@ fn log_format_color(
     )
 }
 
-fn set_logging_to_files(logger: Logger, log_dir: &Path) -> Logger {
+#[derive(Debug)]
+struct FileLogConfig<'a> {
+    dir: &'a Path,
+    keep_compressed: usize,
+    keep_uncompressed: usize,
+    rotate_age: Age,
+    rotate_size: Option<u64>,
+}
+
+impl<'a> FileLogConfig<'a> {
+    fn with_env(dir: &'a Path, messages: &mut Vec<(log::Level, String)>) -> Self {
+        let mut config = Self::new(dir);
+
+        if let Ok(uncompressed) = env::var("LOG_FILES_UNCOMPRESSED") {
+            match uncompressed.parse::<usize>() {
+                Ok(n) => config.keep_uncompressed = n,
+                Err(e) => messages.push((log::Level::Error, format!(
+                    "LOG_FILES_UNCOMPRESSED ({uncompressed}) doesn't contain a valid nonnegative integer: {e}"
+                ))),
+            }
+        }
+        if let Ok(compressed) = env::var("LOG_FILES_COMPRESSED") {
+            match compressed.parse::<usize>() {
+                Ok(n) => config.keep_compressed = n,
+                Err(e) => messages.push((log::Level::Error, format!(
+                    "LOG_FILES_COMPRESSED ({compressed}) doesn't contain a valid nonnegative integer: {e}"
+                ))),
+            }
+        }
+        if let Ok(age) = env::var("LOG_ROTATE_AGE") {
+            match age.to_ascii_lowercase().as_str() {
+                "day" => config.rotate_age = Age::Day,
+                "hour" => config.rotate_age = Age::Hour,
+                _ => messages.push((
+                    log::Level::Error,
+                    format!("LOG_ROTATE_AGE ({age}) is neither DAY nor HOUR nor unset"),
+                )),
+            }
+        }
+        if let Ok(size) = env::var("LOG_ROTATE_SIZE") {
+            match size.parse::<u64>() {
+                Ok(n) => config.rotate_size = Some(n),
+                Err(e) => messages.push((
+                    log::Level::Error,
+                    format!(
+                        "LOG_ROTATE_SIZE ({size}) doesn't contain a valid positive integer: {e}"
+                    ),
+                )),
+            }
+        }
+
+        config
+    }
+
+    fn new(dir: &'a Path) -> Self {
+        FileLogConfig {
+            dir,
+            keep_compressed: 10,
+            keep_uncompressed: 1,
+            rotate_age: Age::Day,
+            rotate_size: None,
+        }
+    }
+}
```

Comment:
By using `structopt` you could avoid most of the parsing code.

Reply:
I considered that, but `structopt` doesn't allow me to communicate detect per env without writing as much boilerplate as there already is AFAICT.